docs/architecture.md

```diff
@@ -0,0 +1,35 @@
+# OpenShift Hive Architecture
+
+OpenShift Hive is a Kubernetes operator built to provide a declarative API to provision, configure, reshape, and deprovision OpenShift clusters at scale. Hive's API is a set of Kubernetes Custom Resource Definitions, served by the Kubernetes apiserver.
+
+Hive leverages the [OpenShift Installer](https://github.com/openshift/installer) to perform cluster provisioning and deprovisioning.
+
+
+## Components
+
+ 1. **Hive Operator**: The first Hive pod to run is the hive-operator which is essentially a deployer. It manages the Kubernetes Deployments for other components to ensure they are properly configured and running. It also handles migrations and cleanup and anything else that might require programatic logic in a long term Hive deployment. Hive's operator reconciles on a HiveConfig CRD which provides some configuration options for Hive itself.
+ 1. **Hive Admission**: Hive Admission is a small stateless HTTP server used for CRD webhook validation. It's just is just to approve or deny creation or updates to our core CRDs.
+ 1. **Hive Controllers**: The core Hive controllers which reconcile all CRDs.
+
+## How Hive Works
+
+![Hive Architecture](hive-architecture.png "Hive Architecture")
+
+API callers create ClusterDeployments which contain the desired state for the cluster they would like to provision. Hive was intended to work in a multi-tenant environment, so ClusterDeployments and their associated secrets can be created in separate namespaces. This allows for protecting access to data with standard Kubernetes RBAC.
+
+ClusterDeployments define the OpenShift release image to install, either directly on the ClusterDeployment, or by linking to a pre-existing ClusterImageSet. A ClusterImageSet is just a small CRD which links an OpenShift release image to a corresponding Hive image that work with that particular installer version.
+
+Once a ClusterDeployment is created, ignoring a few optional pre-install steps Hive can optionally automate, an install pod will be launched. The install pod consists of two containers, the installer extracted from the specified release image, and a Hive installmanager sidecar container. The openshift-install binary is copied from the install container into the installmanager container where we can execute it and upload required artifacts during and after the install is completed.
```

Comment:
This sounds odd ` Once a ClusterDeployment is created, ignoring a few optional pre-install steps Hive can optionally automate, an install pod will be launched`. Need to rephrase.